### PR TITLE
DM-10156: Replace all uses of Calib with PhotoCalib

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: src/UnitSystem.cc [hunk 1]
```diff
@@ -29,57 +29,49 @@
...
+namespace lsst {
+namespace meas {
+namespace modelfit {
 
-UnitSystem::UnitSystem(afw::geom::SpherePoint const & position,
-                       std::shared_ptr<const lsst::afw::image::Calib> calibIn,
-                       double flux) {
-    auto scale = 1.0*lsst::afw::geom::arcseconds;
+UnitSystem::UnitSystem(afw::geom::SpherePoint const& position,
+                       std::shared_ptr<const lsst::afw::image::PhotoCalib> photoCalib_, double flux) {
```
Comment (kfindeisen) [inline src/UnitSystem.cc]: The underscore looks a little weird here. Maybe rename the internal variable to `_photoCalib`, per the [style guide](https://developer.lsst.io/cpp/style.html#private-class-variables-and-methods-must-have-underscore-prefix)?

File: src/UnitSystem.cc [hunk 2]
```diff
@@ -29,57 +29,49 @@
...
 }
 
-std::shared_ptr<const lsst::afw::image::Calib> UnitSystem::validateCalib(
-    std::shared_ptr<const lsst::afw::image::Calib> calib_) {
-    if (calib_->getFluxMag0().first == 0.0){
-        return getDefaultCalib();
-    } else{
-        return calib_;
+std::shared_ptr<const lsst::afw::image::PhotoCalib> UnitSystem::validatePhotoCalib(
+        std::shared_ptr<const lsst::afw::image::PhotoCalib> photoCalib_) {
```
Comment (kfindeisen) [inline src/UnitSystem.cc]: Another underscore in the API.

Comment (kfindeisen) [reply src/UnitSystem.cc]: Consider a `using lsst::afw::image::PhotoCalib;` declaration at the head of the file to make the code a bit less verbose.

File: tests/test_projectedLikelihood.py [hunk 3]
```diff
@@ -81,21 +81,20 @@ def setUp(self):
...
-        # setup ideal exposure0: uses fit Wcs and Calib, has delta function PSF
+        # setup ideal exposure0: uses fit Wcs and PhotoCalib, has delta function PSF
         scale0 = 0.2*lsst.afw.geom.arcseconds
         self.crpix0 = lsst.afw.geom.Point2D(0, 0)
         wcs0 = lsst.afw.geom.makeSkyWcs(crpix=self.crpix0,
                                         crval=self.position,
                                         cdMatrix=lsst.afw.geom.makeCdMatrix(scale=scale0))
-        calib0 = lsst.afw.image.Calib()
-        calib0.setFluxMag0(10000)
+        photoCalib0 = lsst.afw.image.PhotoCalib(10)
```
Comment (kfindeisen) [inline tests/test_projectedLikelihood.py]: Is this a change of units?

File: src/UnitSystem.cc (hunk 1 above)
Comment (parejkoj) [reply src/UnitSystem.cc]: Except that it's a `struct`, so the internal variable is `public`.

File: src/UnitSystem.cc (hunk 2 above)
Comment (parejkoj) [reply src/UnitSystem.cc]: I removed the `lsst::` from it, which helps.

File: tests/test_projectedLikelihood.py (hunk 3 above)
Comment (parejkoj) [reply tests/test_projectedLikelihood.py]: Sort of: `PhotoCalib(10) != Calib(10000)`, for any definition of `==`. However, the actual value of the photometric calibration is not important in this test, just self-consistency.

File: src/UnitSystem.cc (hunk 1 above)
Comment (kfindeisen) [reply src/UnitSystem.cc]: Ah. The constructor must have thrown me off. I assume you won't consider calling both `photoCalib`?

**Pull request merged**